src/tests/tokens/nf-token.test.ts

```diff
@@ -397,4 +397,15 @@ spec.test('throws when trying to burn non existant NFT', async (ctx) => {
   const id1 = ctx.get('id1');
 
   await ctx.reverts(() => nftoken.instance.methods.burn(id1).send({ from: owner }), '003002');
+});
+
+spec.test('safeTransfer does not call onERC721Received to constructing contract', async (ctx) => {
+  const sendsToSelfOnConstruct = await ctx.deploy({ 
+    src: './build/sends-to-self-on-construct.json',
+    contract: 'SendsToSelfOnConstruct',
+  });
+
+  const receipt = sendsToSelfOnConstruct.receipt;
+  console.log(receipt.events.Received());
+  ctx.not(receipt.events.Received, undefined); // I want to confirm here that there is only one event (the mint(), and not the safeTransferFrom)
```

Comment:
You have all the emmited events here. So even if you are expecting same kind of event. Lets say `Transfer` you will have an array of `Transfer` events here and can check if there is only one and how it is constructed.